src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -234,8 +234,17 @@ def _migrate_service_principals(
 
         execution_result = []
         for spn in sp_list_with_secret:
+            if spn.permission_mapping.default_network_action != "Allow":
+                logger.warning(
```

Comment:
does this mean we still create the storage credential with service principal, and then will create another storage credential with managed identity? shouldn't it be either/or?

Reply:
@nkvuong : Thanks for your input - once more. 

I gave it a though:
1. I am not 100% sure that storage credentials with service principals that access storage behind firewalls will **always** be an issue. It's probably likely, though, it could not be the case. 
2. At least we want to notify the user about the risk of (1)
3. Additionally, we should nudge users to use access connectors as it avoids (1) and is a Databricks best practice

I updated the code to prompt the user once more before migrating **all** service principals if at least one service principal is used to access a storage accounts behind firewall. Additionally, we still warn the user with details on which service principal access which storage account behind a firewall, if the user whises to continue.

Reply:
On the either/or, we decided to support creating both storage credentials for service principals and access connectors to provide flexibility to the user. Though, we (should) make it clear that we prefer access connectors.

Reply:
my thinking is that you need to specify a storage credential per external location (~ a storage account). So the external location migration logic needs to pick one, our preference would be to pick the one with an access connector

Reply:
the logic here is a bit fragile - https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/azure/locations.py#L65

Reply:
let's cover that in #1426 as that PR creates the storage credentials for the access connectors with managed identities